### DISCUSSION TITLE
MAISI Mask Autoencoder

### DISCUSSION THREAD

**Discussion by cbe135** [Project-MONAI/tutorials#2026] [Q&A] [answered]
During the implementation of our own Mask Autoencoder, we encountered two questions.
Specifically, our dataset masks have the characteristic of input channel of 1 and output channel of 1.
Our question relates to the different input channel and output channel setting used in pretraining the provided model weights.
The output channel count of 128 makes sense as there are 128 possible option choices for organs and disease phenomena.
In terms of the input channel count, we were wondering why the input channel was 7?
Also, in terms of fine tuning it to our dataset, we were thinking of averaging/modifying the first layer and last layer of the model into the desired input or output. Are there other recommended suggestions?
Thank you.

**Comment by Can-Zhao** [comment]
@guopengf

**Comment by guopengf** [comment, marked as answer]
Hi @cbe135, the input channel of the mask autoencoder is 8 (see https://github.com/Project-MONAI/tutorials/blob/8b90a16d82a463551d81dd4da200f3d406c6843e/generation/maisi/configs/config_maisi3d-rflow.json#L94). 
This is mainly because we use binary representation to encode the input mask, which saves memory. 8 channels can represent 2**8 (0~255) labels. Each channel represents a bit (see the following function).  https://github.com/Project-MONAI/tutorials/blob/8b90a16d82a463551d81dd4da200f3d406c6843e/generation/maisi/scripts/utils.py#L175-L190
For example, label 1 is encoded as [0, 0, 0, 0, 0, 0, 0, 1].

For your use case, is the label of your dataset covered in the pre-defined [label dict](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/configs/label_dict.json)?

**Comment by cbe135** [comment]
Hi @guopengf, thank you.
That makes sense.
Our mask target is hypo tumor so it is not on the list.
Would it make sense to use one of the dummies?
Does it matter which dummy is used?
Looking specifically at the [label dict](https://github.com/Project-MONAI/tutorials/blob/main/generation/maisi/configs/label_dict.json), is there any specific logic on how the different labels are ordered?
Thank you.

**Reply by guopengf** [reply]
The order of labels does not matter, so you can use any dummy label.

**Comment by mar-cry** [comment]
Hello @guopengf , my dataset has three categories that are not included in the 132 predefined categories in the label_dict, so I mapped these three categories to dummy6, dummy7, and dummy8. After converting them into the input format compatible with the Mask Autoencoder, encoding the input mask, and then decoding to reconstruct, I found that the reconstructed result differs greatly from the input. Is this normal? Here are illustrative examples of my input and output.
<img width="837" height="592" alt="QQ_1757493096727" src="https://github.com/user-attachments/assets/387e9f20-bac9-49fc-a0ee-f2b41b60b788" />
<img width="833" height="606" alt="QQ_1757493104368" src="https://github.com/user-attachments/assets/97b0f8f3-f83a-4a04-98f9-08085f800fb3" />

**Reply by guopengf** [reply]
Have you trained the AE with dummy labels? If not, this result should be expected. Please try to fine-tune AE with your dataset.

**Reply by mar-cry** [reply]
I don't fine-tune the mask autoencoder on our dataset. I notice that the input and output channels of mask autoencoder are different from the common image autoencoder, is there any difference between the training pipeline of two autoencoders, such as the training loss?
